hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/InvocationFuture.java

```diff
@@ -144,9 +146,21 @@ protected Object resolve(Object unresolved) {
     /**
      * Any Object is considered a value-type response, even exceptions.
      * The only exception in the above rule, is responses of type {@link ErrorResponse}, which are thrown instead of returned.
+     *
+     * RU_COMPAT_3_10 For compatibility reasons, when the cluster version is less than 3.11, we keep treating Throwables, as
+     * proper Exceptions & not values.
      */
     private boolean isErroneousState(Object state) {
-        return state instanceof ErrorResponse;
+        Version version = invocation.context.clusterService.getClusterVersion();
+        return state instanceof ErrorResponse || (version.isLessThan(V3_11) && state instanceof Throwable);
+    }
+
+    private Throwable unwrap(Object error) {
+        try {
+            return ((ErrorResponse) error).getCause();
+        } catch (Exception e) {
```

Comment:
I'm a bit confused about the exception catching here.  The getCause can't fail.. So either error is null, or it isn't an ErrorResponse; both cases can be handled with a check.

Reply:
The Exception handling is for class-cast in case its not an ErrorResponse and its a Throwable. I found it convenient, but if it's unclear, I can switch to a plain instanceof check.